### PR TITLE
Add methods for exposing timestamps of events and timeout message.

### LINKED ISSUE #78: Consider adding a Time() method to ProcessTimeoutScope
I think it's pretty reasonable to want to know the time that a timeout is scheduled to occur. However, this hasn't been done yet as the argument is that the message itself should contain the time if it is relevant to the domain. This is certainly true for commands and events, but I wonder if this is just an annoyance for timeouts, which are essentially an implementation detail of the process they belong to.

### PR DESCRIPTION
**Pull request by jmalloc**
Fixes #78.

### CODE REVIEW

File: docs/adr/0011-message-timing-information.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+Accepted
+
+## Context
+
+We need to decide whether message timing information should be exposed via the
+API. Timing information refers to important points in time throughout the
+lifecycle of a message.
+
+The initial rationale for *not* exposing these timestamps was that any business
+logic that depends on time in some way should include any time related
```
Comment (danilvpetrov) [inline docs/adr/0011-message-timing-information.md]: should `time related` be hyphenated here?

File: docs/adr/0011-message-timing-information.md [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+The time at which a timeout message is scheduled to be handled is a fundamental
+property of the timeout concept.
+
+By using a timeout message the process is declaring that there is some
+time-based component to its logic. It seems like an unnecessary hurdle to
+require the application developer to include the scheduled time in the message
+when the engine must necessarily keep track of this time anyway.
+
+**We have decided to include a `ScheduledFor()` method on `ProcessTimeoutScope`.
```
Comment (koden-km) [inline docs/adr/0011-message-timing-information.md]: No closing **bold**?

File: docs/adr/0011-message-timing-information.md [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+## Consequences
+
+As a result of this ADR it is easier for application developers to implement
+logic that depends on time. This is especially true when new logic is added that
+is triggered by domain logic that did not already have some time-based
+component.
+
+As a result, engine implements MUST record the time at which events occur. This
```
Comment (koden-km) [inline docs/adr/0011-message-timing-information.md]: Did you mean "implementors" or "implementations"?
> engine implements MUST

Comment (jmalloc) [reply docs/adr/0011-message-timing-information.md]: I meant `implementations`... I also didn't mean to capitalize MUST in this case.

File: docs/adr/0011-message-timing-information.md (hunk 1 above)
Comment (jmalloc) [reply docs/adr/0011-message-timing-information.md]: I'll revisit this terminology throughout the ADR, it's probably a bit inconsistent atm.

File: docs/adr/0011-message-timing-information.md (hunk 2 above)
Comment (jmalloc) [reply docs/adr/0011-message-timing-information.md]: Fixed.

**Comment by jmalloc** [comment]
I've updated the wording of the ADR somewhat.

File: docs/adr/0011-message-timing-information.md [hunk 4]
```diff
@@ -23,30 +24,29 @@ timestamps of interest, and the decisions made in each case.
...
 
 The time at which an event is recorded is a fundamental property of the event
 itself. Put another way, every event occurs at some time regardless of whether
-the domain has any time-based logic.
+the domain is time-based.
 
-Furthermore, the time at which event occurs may be relevant to some domain logic
-that is *triggered* by the event, even if the aggregate that *produced* the event
-has no time-based logic.
+Furthermore, the time at which event occurs may be relevant to some ancillary
```
Comment (koden-km) [inline docs/adr/0011-message-timing-information.md]: This might be better as *an event* or *the event*?
> the time at which event occurs may

Comment (jmalloc) [reply docs/adr/0011-message-timing-information.md]: Added missing `the`.

**Pull request merged**